### PR TITLE
Bb 13 event card component

### PR DESCRIPTION
**Pull request by DylanHopkins21**
## Describe your changes:
Created the EventCard component. Takes in all the props described in the Jira Issue ticket. Additionally, it takes in the owner boolean, displaying the owner badge if true. 

## Proof that your changes work (include screenshots if frontend):
image

## Jira Issue ticket number and link:
BB-13
https://bearbuddies.atlassian.net/jira/software/projects/BB/boards/1?selectedIssue=BB-13

## Checklist before requesting a review
- [*] I have performed a self-review of my code
- [*] I have DM'd the group chat that I have made a PR and need reviewers.
- [*] I haved moved the Jira issue to "Under Review"
- [*] I have run `npx eslint filename.js` on all JavaScript files I've changed in this PR and fixed all displayed errors

### CODE REVIEW

File: src/components/EventCard/EventCard.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        joined: Boolean,
+        owner: Boolean,
+    }
+    let buttText;
+    if (joined) {
+        buttText = "Happening Soon!";
+    } else {
+        buttText = "Join!"
+    }
+    let badge;
```
Comment (thomaspwang) [inline src/components/EventCard/EventCard.js:27]: use const instead of let

File: src/components/EventCard/EventCard.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        authorPic: String,
+        maxCap: Number,
+        currCap: Number,
+        location: String,
+        time: String,
+        description: String,
+        joined: Boolean,
+        owner: Boolean,
+    }
+    let buttText;
```
Comment (thomaspwang) [inline src/components/EventCard/EventCard.js:21]: use const

File: src/components/EventCard/EventCard.js (hunk 1 above)
Comment (thomaspwang) [inline src/components/EventCard/EventCard.js:27]: use const

File: src/components/EventCard/EventCard.js [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    if (joined) {
+        buttText = "Happening Soon!";
+    } else {
+        buttText = "Join!"
```
Comment (thomaspwang) [inline src/components/EventCard/EventCard.js]: as funny as it is name buttText to buttonText

File: src/components/EventCard/EventCard.js [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    }
+    let badge;
+    if (owner) {
+        badge = <OwnerBadge />
+    } else {
+        badge = <div />
+    }
```
Comment (thomaspwang) [inline src/components/EventCard/EventCard.js]: add spaces between logic blocks so easier to read

- space between let buttText
- space before let badge

File: src/components/EventCard/EventCard.js
Comment (thomaspwang) [inline src/components/EventCard/EventCard.js:1]: for icon SVGs include those in the public folder and then import them in

Comment (thomaspwang) [inline src/components/EventCard/EventCard.js:1]: try to style it more similarly to the figma 
- image larger
- (around) equal whitespace at top / bottom of card
- slightly larger text and icons
- smaller capacity logo and text (should be the same size as the card text)
- smaller title probably

image

Comment (thomaspwang) [inline src/components/EventCard/EventCard.js:1]: Make sure you're using your global styles for standard text, for example, the title and normal text.

currently it's not DM sans but it should be

File: src/components/EventCard/EventCard.js [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+function OwnerBadge() {
+    return (
+        <div className={styles.iconText}>
+            <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24">
+                <path fill="#7aa3ff" d="m12 23l-3-3H5q-.825 0-1.413-.588T3 18V4q0-.825.588-1.413T5 2h14q.825 0 1.413.588T21 4v14q0 .825-.588 1.413T19 20h-4l-3 3Zm0-11q1.45 0 2.475-1.025T15.5 8.5q0-1.45-1.025-2.475T12 5q-1.45 0-2.475 1.025T8.5 8.5q0 1.45 1.025 2.475T12 12Zm0 8.2l2.2-2.2H19v-1.15q-1.35-1.325-3.138-2.087T12 14q-2.075 0-3.863.763T5 16.85V18h4.8l2.2 2.2Z"/>
+            </svg>
+            <p id={styles.badgeText}>Your Group</p>
+        </div>
+    )
+}
```
Comment (thomaspwang) [inline src/components/EventCard/EventCard.js]: generally it's syntactic practice to have secondary components above the export default component, so move it up

File: src/styles/globals.css
Comment (thomaspwang) [inline src/styles/globals.css:1]: are you sure you should include this file in this PR? I thought this is in another PR.

**Review by thomaspwang — CHANGES_REQUESTED** [review]
Looks great! Just a few minor changes and it should be pog

File: src/components/EventCard/EventCard.js (hunk 3 above)
Comment (DylanHopkins21) [reply src/components/EventCard/EventCard.js]: Fixed

File: src/components/EventCard/EventCard.js
Comment (DylanHopkins21) [reply src/components/EventCard/EventCard.js:1]: image

Made p much every change, I think the capacity stuff is close to the figma already

File: src/components/EventCard/EventCard.js (hunk 4 above)
Comment (DylanHopkins21) [reply src/components/EventCard/EventCard.js]: Fixed

image

File: src/components/EventCard/EventCard.js
Comment (DylanHopkins21) [reply src/components/EventCard/EventCard.js:1]: They are now all components that are imported in from the public folder

Comment (DylanHopkins21) [reply src/components/EventCard/EventCard.js:1]: Made the change in globals.css, should now be solid for everyone's work from here on out

File: src/components/EventCard/EventCard.js (hunk 5 above)
Comment (DylanHopkins21) [reply src/components/EventCard/EventCard.js]: image

Moved

File: src/styles/globals.css
Comment (DylanHopkins21) [reply src/styles/globals.css:1]: Included because I needed to fix the DM Sans that will affect everyone

File: src/components/EventCard/EventCard.js
Comment (thomaspwang) [reply src/components/EventCard/EventCard.js:1]: amazing, and looks great.

**Review by thomaspwang — APPROVED** [review]
Woo! Looks great to me. Thanks for clarifying your changes by responding to comments; they're super helpful when reviewing,

**Review by LeonJia1 — APPROVED** [review]
looks good to me

**Comment by LeonJia1** [comment]
also reorder ur imports >:(

File: src/components/EventCard/EventCard.js [hunk 6]
```diff
@@ -0,0 +1,93 @@
+import 'bootstrap/dist/css/bootstrap.min.css';
+import Badge from '../../../public/badge';
+import Clock from '../../../public/clock';
+import Location from '../../../public/location';
+import Paragraph from '../../../public/paragraph';
+import Person from '../../../public/person';
+import styles from './EventCard.module.css';
+import { Button } from 'react-bootstrap';
+import React from "react";
+import Image from 'next/image'
```
Comment (LeonJia1) [inline src/components/EventCard/EventCard.js]: organize imports!

**Pull request merged**